### PR TITLE
67 implement smartcar frontcam

### LINKED ISSUE #67: Implement smartcar front camera view on mobile app
As a user, I want to get a view of my car from the front so that I am able to see where the car is heading, especially while initiating a parking by going forward. 

Acceptance criteria
- [x] The hood of the car, and both corners of the hood, with obstacles in close proximity visible to be detected through the cam feed.

### PR DESCRIPTION
**Pull request by adityak714**
For now, the app is able to show a front camera livestream with decent speed, and color-matched to the mod. Upcoming plan for later during the sprint:

- Setup cam#2 with birdseye view in #64. Need to add an extra camera (change in vehicle_config.json)

**NOTE: All team members need SMCE to be updated to a later version to handle multiple camera implementation further on.**

- The 2 buttons that were added in Android are to be implemented more in depth during the first half of sprint3, where they will be mapped to show the broadcast in the camera view based on which button is clicked (change MainActivity.java and activity_main.xml)
- Shift the location of the front camera up vertically, up to the level of the dashboard, and potentially back so that a more realistic POV-angle can be seen in the app (change in godot mod)

### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Code change tested manually if not covered by automated tests
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change

### CODE REVIEW

**Comment by adityak714** [comment]
> I suggest taking a look at what your friends are doing at [DIT113-V22/group-08#63](https://github.com/DIT113-V22/group-08/pull/63) ;) uploading the .pck will very much slowdown the git repo over time

@RuthgerD Ok ... I went through their pull request, but I don't understand this line here:

DIT113-V22/group-08#63
```image: barichello/godot-ci:mono-latest```

**Comment by adityak714** [comment]
> This means it uses that docker image for the ci run instead of the default ubuntu one; in short it provides the godot binary.
> 
> I suggest making a separate issue for this if you want to try implementing it

Alright got it, thanks for the suggestion king 👑

**Review by sh4r10 — CHANGES_REQUESTED** [review]
The buttons for the camera views should be removed from this branch, not only do they take up too much space they are also unnecessary for the issue itself. A button will be later on which takes care of switching between the two cameras, but as of now, this is not needed.

**Comment by um33** [comment]
looks fine to me because  i did some work on this branch with Aditya so everything is understandable to me.

**Comment by Gabri1998** [comment]
The Car-FrontCamera almost fixed except that we need to do some identification in the code  between the FrontCamera and the Birdseye view camera . other than that It works as it supposed to be

**Comment by adityak714** [comment]
> The buttons for the camera views should be removed from this branch, not only do they take up too much space they are also unnecessary for the issue itself. A button will be later on which takes care of switching between the two cameras, but as of now, this is not needed.

Commit 88fb92e should fix this. @um33 and me removed the buttons here.

**Comment by arrestoteles** [comment]
@adityak714 @um33 "P" button doesn't work - is it supposed to be implemented in this branch? Good work with the frontcam! 📸

File: arduino/smartcar/smartcar.ino [hunk 1]
```diff
@@ -58,7 +66,7 @@ const int TURNING_INCREMENT = 10;
 const int FORWARD_SPEED_LIMIT = 150;
 const int BACKWARD_SPEED_LIMIT = -50;
 const int MAX_STEERING_ANGLE = 60;
-const auto ONE_SECOND = 1000UL;
+const auto oneSecond = 1000UL;
```
Comment (arrestoteles) [inline arduino/smartcar/smartcar.ino:69]: Constants should be capitalized no?

**Comment by um33** [comment]
> @adityak714 @um33 "P" button doesn't work - is it supposed to be implemented in this branch? Good work with the frontcam! 📸
> 
> Also, we should change the acceptance criteria for issue #67

P button just not working on this branch but it is working on the master.

**Comment by um33** [comment]
now camera is not working on my branch after updating the SMCE new version, happened with one other group mate as well @RuthgerD

**Comment by RuthgerD** [comment]
If you're using a custom config for it you need to update some keys as the way it works changed :) check the slack announcement thread for info

**Review by um33 — DISMISSED** [review]
This is good, I have tested it on my Android emulator and the cam feed is working as normal now. Also tested on another computer's emulator, and in method calls for autopark and retrieve(), cam feed does not stop anymore. New setup for board_config and vehicle_config, so all in team need to update smce 1.3.4 and birdseye cam can be started, if time allows

**Review by sh4r10 — APPROVED** [review]
The branch covers the acceptance criteria of the related issue, the changes have been testing manually and no problems were encoutered. This is ready to be merged.

**Pull request merged**